Subject: Detect rst filetype using contents of file
 Inspect the first 5 lines of the file to see if there's a leading ".. "
 or stand-alone "restindex".  If either are found, set the filetype to
 rst.
Author: James Vega <jamessan@debian.org>
Bug-Debian: 382541
Forwarded: 2006-08-12

Index: b/runtime/scripts.vim
===================================================================
--- a/runtime/scripts.vim
+++ b/runtime/scripts.vim
@@ -322,6 +322,14 @@
   elseif s:line1 =~ 'exec\s\+\S*scheme' || s:line2 =~ 'exec\s\+\S*scheme'
     set ft=scheme
 
+  " rst files
+  elseif s:line1 =~ '^\.\.\s\|^\s*restindex\s*$'
+	\ || s:line2 =~ '^\.\.\s\|^\s*restindex\s*$'
+	\ || s:line3 =~ '^\.\.\s\|^\s*restindex\s*$'
+	\ || s:line4 =~ '^\.\.\s\|^\s*restindex\s*$'
+	\ || s:line5 =~ '^\.\.\s\|^\s*restindex\s*$'
+    set ft=rst
+
   " Git output
   elseif s:line1 =~ '^\(commit\|tree\|object\) \x\{40\}\>\|^tag \S\+$'
     set ft=git
